Description: keep hardening flags and system libraries out of icu-config
Origin: upstream, http://www.icu-project.org/trac/changeset/34745
Bug: http://bugs.icu-project.org/trac/ticket/6102
Bug-Debian: http://bugs.debian.org/759792

Index: icu/source/config/Makefile.inc.in
===================================================================
--- icu.orig/source/config/Makefile.inc.in
+++ icu/source/config/Makefile.inc.in
@@ -1,6 +1,6 @@
 ## -*-makefile-*-
 #******************************************************************************
-#   Copyright (C) 1999-2012, International Business Machines
+#   Copyright (C) 1999-2013, International Business Machines
 #   Corporation and others.  All Rights Reserved.
 #******************************************************************************
 # This Makefile.inc is designed to be included into projects which make use
@@ -115,7 +115,7 @@ ICULIBS_ICUIO = -l$(ICUPREFIX)io$(ICULIB
 ICULIBS_OBSOLETE = -l$(ICUPREFIX)obsolete$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
 ICULIBS_LAYOUT = -l$(ICUPREFIX)le$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
 ICULIBS_LAYOUTEX = -l$(ICUPREFIX)lx$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
-ICULIBS_BASE = $(LIBS) -L$(libdir)
+ICULIBS_BASE = -L$(libdir)
 
 # for icu-config to test with
 ICULIBS_COMMON_LIB_NAME = ${LIBICU}${COMMON_STUBNAME}${ICULIBSUFFIX}${ICULIBSUFFIX_VERSION}.${SO}
@@ -186,11 +186,11 @@ AR = @AR@
 	ARFLAGS := @ARFLAGS@ $(ARFLAGS)
 #SH#ARFLAGS="@ARFLAGS@ ${ARFLAGS}"
 CC = @CC@
-CFLAGS = @CFLAGS@ 
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@ $(LIBCPPFLAGS) -I$(prefix)/include
+CFLAGS = @UCONFIG_CFLAGS@ 
+CPPFLAGS = @UCONFIG_CPPFLAGS@ -I$(prefix)/include
+CXXFLAGS = @UCONFIG_CXXFLAGS@
 CXX = @CXX@
-CXXFLAGS = @CXXFLAGS@ 
 DEFAULT_MODE = @DATA_PACKAGING_MODE@
 DEFS = @DEFS@ 
 FFLAGS = @FFLAGS@
@@ -201,7 +201,7 @@ INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM  =  @INSTALL_PROGRAM@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
-LDFLAGS = @LDFLAGS@ $(RPATHLDFLAGS)
+LDFLAGS = $(RPATHLDFLAGS)
 LIBS = @LIBS@ 
 LIB_M = @LIB_M@
 LIB_VERSION = @LIB_VERSION@
Index: icu/source/config/icu-config-bottom
===================================================================
--- icu.orig/source/config/icu-config-bottom
+++ icu/source/config/icu-config-bottom
@@ -258,7 +258,7 @@ do
 	    ;;
 
 	--ldflags)
-	    echo $ECHO_N "${LDFLAGS} ${ICULIBS} ${LIBS} ${ECHO_C}"
+	    echo $ECHO_N "${LDFLAGS} ${ICULIBS} ${ECHO_C}"
 # $RPATH_LDFLAGS
 	    ;;
 
